internal/services/controllers/v1/task/controller.go

```diff
@@ -819,7 +819,60 @@ func (tc *TasksControllerImpl) handleProcessUserEventTrigger(ctx context.Context
 		})
 	}
 
-	tasks, dags, err := tc.repov1.Triggers().TriggerFromEvents(ctx, tenantId, opts)
+	result, err := tc.repov1.Triggers().TriggerFromEvents(ctx, tenantId, opts)
+
+	if err != nil {
+		return fmt.Errorf("could not trigger tasks from events: %w", err)
+	}
+
+	eventTriggerOpts := make([]tasktypes.CreatedEventTriggerPayloadSingleton, 0)
+	eventSeenAt := time.Now()
+
+	for _, event := range result.Events {
```

Comment:
(see note below first) I think this blows up much faster than we think -- these buffers can easily saturate to 1000 events + tasks at peak, so 1 million iterations?